lib/gameState/CGameState.cpp

```diff
@@ -904,7 +904,7 @@ void CGameState::initTowns()
 		}
 		//init spells
 		vti->spells.resize(GameConstants::SPELL_LEVELS);
-
+		vti->possibleSpells -= SpellID::PRESET;
```

Comment:
It is not harmful, but not sure how SpellID::PRESET can even appear in towns.
It is used to indicate starting spells of a hero.

Reply:
I saw that editor uses it as some kind of placeholder, probably related to that

Reply:
Yes, I used it also for marking that town possible spells had been customized, so editor can detect that, otherwise I would have to compare town possible spells with allowed spells for differences to know if spells were customized. 
May create separate const specific for towns if that one was meant bo be hero only

Reply:
Should be fine. At the moment it is only used for heroes, but using it for towns if needed should also be fine.

Personally I don't like this approach with fake spells, but don't see any clean solution either. And it is not like this weird approach was introduced in this PR.